tests/cluster/tests/20-half-migrated-slot.tcl

```diff
@@ -0,0 +1,153 @@
+# Tests for fixing migrating slot at all stages:
+# 1. when migration is half inited on "migrating" node
+# 2. when migration is half inited on "importing" node
+# 3. migration inited, but not finished
+# 4. migration is half finished on "migrating" node
+# 5. migration is half finished on "importing" node
+source "../tests/includes/init-tests.tcl"
+
+test "Create a 2 nodes cluster" {
+    create_cluster 2 0
+}
+
+test "Cluster is up" {
+    assert_cluster_state ok
+}
+
+set cluster [redis_cluster 127.0.0.1:[get_instance_attrib redis 0 port]]
+catch {unset nodefrom}
+catch {unset nodeto}
+proc reset_cluster {} {
+    uplevel 1 {
+        $cluster refresh_nodes_map
+        array set nodefrom [$cluster masternode_for_slot 609]
+        array set nodeto [$cluster masternode_notfor_slot 609]
+    }
+}
+
+proc fix_cluster {} {
+    uplevel 1 {
+        set code [catch {
+            exec ../../../src/redis-cli --cluster fix $nodefrom(addr) << yes
+        } result]
+        if {$code != 0 && $::verbose} {
+            puts $result
+        }
+        assert {$code == 0}
+        assert_cluster_state ok
+        wait_for_condition 1000 10 {
+            [catch {exec ../../../src/redis-cli --cluster check $nodefrom(addr)} _] == 0
+        } else {
+            fail "Cluster could not settle with configuration"
+        }
+    }
+}
+
+reset_cluster
+
+test "Set key" {
+    # 'aga' key is in 609 slot
+    $cluster set aga xyz
+}
+
+test "Key is accessible" {
+    assert {[$cluster get aga] eq "xyz"}
+}
+
+test "Fix doesn't need to fix anything" {
+    set code [catch {exec ../../../src/redis-cli --cluster fix $nodefrom(addr) << yes} result]
+    assert {$code == 0}
+}
+
+test "Key is accessible" {
+    assert {[$cluster get aga] eq "xyz"}
+}
+
+test "Half init migration in 'migrating'" {
+    $nodefrom(link) cluster setslot 609 migrating $nodeto(id)
+}
+
+test "Fix cluster" {
+    fix_cluster
```

Comment:
The common convention with other cluster tests (and the Redis test suite in general) is to create larger tests that consist of multiple assertions. I think it would be a good idea for these tests to follow this convention as well for uniformity (pros and cons are arguable).